Feature/cluster support with spring session

IgorNB opened the pull request:
Cluster support with spring session added.

- Before this MR tesler platform in cluster mode produced error (see 2 node [scenario](https://youtu.be/iHdriuhoEu8))

- After this MR one can choose any spring-session provider [(jdbc, redis and so on)](https://spring.io/projects/spring-session) and work in cluster mode (see [scenario](https://youtu.be/2K2jZy6iGpY) for [tesler-doc](https://github.com/tesler-platform/tesler-doc/pull/22) exmaple application. Includes configuring spring-session-jdbc and working with 2 nodes)

Commit:
Cluster support added with spring-session (see tesler-doc for example with spring-session-jdbc)

Commit:
remove unused imports

Dergash commented on tesler-core/src/main/java/io/tesler/core/util/session/impl/SessionServiceImpl.java:
```diff
@@ -219,7 +216,7 @@ public String getSessionIpAddress() {
 	@Override
 	public Map<String, Boolean> getResponsibilities() {
-		return sessionCache.getResponsibilities(getSessionUser(), getSessionUserRole());
+		return uiService.getResponsibilities(getSessionUser(),getSessionUserRole());
```
Will it affect the performance? It seems the views are cached by separate `UICache` but responsibilities are not.

IgorNB replied:
>>Will it affect the performance?

This methods used session scoped SESSION_CAHCE. 
In cluster mode with spring-session-jdbc this cache was stored in DB (<=> there was no cache), so performace was same.

But we can improve perfomance in cluster mode if we'll use this cache without session scope  (i.e. force it to be stored in RAM on each node like UI_CACHE).

I've done it. Now, methods previously cached with SESSION_CAHCE cached again with USER_CACHE, which is not scoped to session

Trade off here is average RAM size for USER_CACHE (elements are not deleted from cache, when user logs out), but when all users log-in at the same time, then both session scoped and current caches will require same RAM. So max RAM size is the same.
 
 In concrete app using tesler and having lots of users one should consider redefining bean 
 @Primary
 @Bean(name = USER_CACHE)
 public Cache userCache() {}
 with some cache provider supporting memory limits and auto cache eviction (hazelcast, redis and so on), so USER_CACHE is kept as separate cache from UI_CACHE in this commit

abratashev commented:
As I know, `BcStateAwareImpl` store `ClientStorage` in session.
`ClientStorage` is serializable now.
`ClientStorage` contains collection of `BcState`, that represent state of current Business component
`BcState` is also serializable.
But `DataResponseDTO`, that a part of `BcState`, isn't serializable.
As long as I remember, spring jdbc does not allow serialize objects, that not implements `Serializable` interface.
Wouldn't such an implementation lead to an error when trying to create a record on a business component that supports state?

IgorNB commented:
> As I know, `BcStateAwareImpl` store `ClientStorage` in session.
> `ClientStorage` is serializable now.
> `ClientStorage` contains collection of `BcState`, that represent state of current Business component
> `BcState` is also serializable.
> But `DataResponseDTO`, that a part of `BcState`, isn't serializable.
> As long as I remember, spring jdbc does not allow serialize objects, that not implements `Serializable` interface.
> Wouldn't such an implementation lead to an error when trying to create a record on a business component that supports state?

It is seems to work fine. Illustrated it at this moment https://youtu.be/2K2jZy6iGpY?t=121

Commit:
Methods previously cached with SESSION_CAHCE now cached again in USER_CACHE which is not scoped to session.

This will give better performance in cluster mode with spring-session-jdbc (<=> no cache, because it stores session data in DB). At the same time for single node mode it will give almost same performance as session scoped cache.

Trade off here is average RAM size for USER_CACHE (elements are not deleted from cache when user logs out), but when all users log-in at the same time, then both session scoped and current caches will require same RAM. So max RAM size is the same.

 In concrete app using tesler and having lots of users one should consider redefining bean
 @Primary
 @Bean(name = USER_CACHE)
 public Cache sessionCache() {}
 with some cache provider supporting memory limits and auto cache eviction (hazelcast and so on), so USER_CACHE is kept as separate cache from UI_CACHE in this commit

Commit:
sessionCache bean and SessionCache classes renamed to userCache and UserCache
